### DISCUSSION TITLE
Does Goravel supports model events / observers?

### DISCUSSION THREAD

**Discussion by zulh-civo** [goravel/goravel#404] [General]
Coming from Rails background, I'm very familiar with Active Record Callbacks (https://guides.rubyonrails.org/active_record_callbacks.html) where I can manipulate model data pre/post CRUD action.

I'm aware that Goravel was built and inspired by Laravel. Not a Laravel developer myself, I did some research and I found something similar in Laravel called Model Events and Observers:

- https://raviyatechnical.medium.com/laravel-model-events-b88165c40892
- https://laravel.com/docs/11.x/eloquent#observers

Just wondering, does Goravel has something similar?

**Comment by krishankumar01** [comment]
Yes, Goravel also supports model events and observers.

https://www.goravel.dev/ORM/getting-started.html#events

**Reply by zulh-civo** [reply]
Thanks!
